### PR TITLE
expression: implement vectorized evaluation for `builtinFloorRealSig`

### PR DESCRIPTION
**Pull request by tsthght**
### What problem does this PR solve? 
expression: implement vectorized evaluation for `builtinFloorRealSig`, for #12105 

### What is changed and how it works?
```
goos: darwin
goarch: amd64
pkg: github.com/pingcap/tidb/expression
BenchmarkVectorizedBuiltinMathFunc/builtinFloorRealSig-VecBuiltinFunc-8         	 1000000	      2168 ns/op	       0 B/op	       0 allocs/op
BenchmarkVectorizedBuiltinMathFunc/builtinFloorRealSig-NonVecBuiltinFunc-8      	  100000	     21896 ns/op	       0 B/op	       0 allocs/op
```

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: expression/builtin_math_vec.go [hunk 1]
```diff
@@ -365,3 +365,22 @@ func (b *builtinPowSig) vecEvalReal(input *chunk.Chunk, result *chunk.Column) er
...
+	}
+	f64s := result.Float64s()
+	for i := 0; i < len(f64s); i++ {
+		if result.IsNull(i) {
+			continue
+		}
+		f64s[i] = math.Floor(f64s[i])
+	}
+	return nil
+
```
Comment (Reminiscent) [inline expression/builtin_math_vec.go]: Maybe we can delete this blank line?

Comment (tsthght) [reply expression/builtin_math_vec.go]: ok

File: expression/builtin_arithmetic_vec.go [hunk 2]
```diff
@@ -34,14 +34,6 @@ func (b *builtinArithmeticDivideDecimalSig) vecEvalDecimal(input *chunk.Chunk, r
 	return errors.Errorf("not implemented")
 }
 
-func (b *builtinArithmeticModIntSig) vectorized() bool {
```
Comment (qw4990) [inline expression/builtin_arithmetic_vec.go]: We don't need to remove this?

Comment (tsthght) [reply expression/builtin_arithmetic_vec.go]: get it

**Comment by qw4990** [comment]
@zz-jason PTAL~

**Pull request merged**